### PR TITLE
Feature/JRA-67 iOS select origin and destination on the map

### PR DESCRIPTION
**Pull request by YawarValeriano**
Added selection for origin and destination points
Added search by address or places
Added markers on home view
Added tests

### CODE REVIEW

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift [hunk 1]
```diff
@@ -98,6 +99,56 @@ class HomeViewController: UIViewController {
...
+
+    }
+
+    @IBAction func continueButtonAction(_ sender: Any) {
+        let position = mapView.camera.target
+        let pos = GMSMarker(position: position)
+
+        switch viewmodel.pointsSelectionStatus {
+        case.pendingOrigin:
+            pos.title = String.localizeString(localizedString: "origin")
```
Comment (Rodrigo-Schar) [inline RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: add this as a constant string

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift [hunk 2]
```diff
@@ -98,6 +99,56 @@ class HomeViewController: UIViewController {
...
+            labelHelper.text = String.localizeString(localizedString: "select-destination")
```
Comment (Rodrigo-Schar) [inline RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: this one too

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift [hunk 3]
```diff
@@ -98,6 +99,56 @@ class HomeViewController: UIViewController {
...
+            pos.icon = UIImage(named: "origin_point")
+            pos.map = mapView
+            viewmodel.origin = pos
+            viewmodel.pointsSelectionStatus = .pendingDestination
+
+        case.pendingDestination:
+            pos.title = String.localizeString(localizedString: "destination")
```
Comment (Rodrigo-Schar) [inline RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: same in here add destination as constant string

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift [hunk 4]
```diff
@@ -98,6 +99,56 @@ class HomeViewController: UIViewController {
         self.mapView.animate(toZoom: zoom)
     }
 
+    @IBAction func backButtonAction(_ sender: Any) {
+        switch viewmodel.pointsSelectionStatus {
+        case.pendingOrigin:
+            return
+        case.pendingDestination:
+            labelHelper.text = String.localizeString(localizedString: "select-origin")
```
Comment (Rodrigo-Schar) [inline RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: and this one too

**Review by jmacboy — APPROVED** [review]

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift (hunk 1 above)
Comment (YawarValeriano) [reply RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: Changed

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift (hunk 2 above)
Comment (YawarValeriano) [reply RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: changed

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift (hunk 3 above)
Comment (YawarValeriano) [reply RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: changed

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift (hunk 4 above)
Comment (YawarValeriano) [reply RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: changed

**Review by Rodrigo-Schar — APPROVED** [review]

**Pull request merged**